frappe/integrations/doctype/ldap_settings/ldap_settings.py

```diff
@@ -15,155 +15,203 @@ def validate(self):
 
 		if not self.flags.ignore_mandatory:
 			if self.ldap_search_string and self.ldap_search_string.endswith("={0}"):
-				connect_to_ldap(server_url=self.ldap_server_url,
+				self.connect_to_ldap(
 					base_dn=self.base_dn,
-					password=self.get_password(raise_exception=False),
-					ssl_tls_mode=self.ssl_tls_mode,
-					trusted_cert=self.require_trusted_certificate,
-					private_key_file=self.local_private_key_file,
-					server_cert_file=self.local_server_certificate_file,
-					ca_certs_file=self.local_ca_certs_file
-				)
+					password=self.get_password(raise_exception=False))
 			else:
 				frappe.throw(_("LDAP Search String needs to end with a placeholder, eg sAMAccountName={0}"))
 
 
-def get_ldap_client_settings():
-	#return the settings to be used on the client side.
-	result = {
-		"enabled": False
-	}
-	settings = frappe.get_doc("LDAP Settings")
-
-	if settings and settings.enabled:
-		result["enabled"] = True
-		result["method"] = "frappe.integrations.doctype.ldap_settings.ldap_settings.login"
-	return result
-
-
-def connect_to_ldap(server_url,
-                    base_dn,
-                    password,
-                    ssl_tls_mode,
-                    trusted_cert,
-                    private_key_file,
-                    server_cert_file,
-                    ca_certs_file):
-	try:
-		import ldap3
-		import ssl
-
-		if trusted_cert == 'Yes':
-			tls_configuration = ldap3.Tls(validate=ssl.CERT_REQUIRED,
-			                              version=ssl.PROTOCOL_TLSv1)
+	def connect_to_ldap(self, base_dn, password):
+		try:
+			import ldap3
+			import ssl
+
+			if self.require_trusted_certificate == 'Yes':
+				tls_configuration = ldap3.Tls(validate=ssl.CERT_REQUIRED,
+											  version=ssl.PROTOCOL_TLSv1)
+			else:
+				tls_configuration = ldap3.Tls(validate=ssl.CERT_NONE,
+											  version=ssl.PROTOCOL_TLSv1)
+
+			if self.local_private_key_file:
+				tls_configuration.private_key_file = self.local_private_key_file
+			if self.local_server_certificate_file:
+				tls_configuration.certificate_file = self.local_server_certificate_file
+			if self.local_ca_certs_file:
+				tls_configuration.ca_certs_file = self.local_ca_certs_file
+
+			server = ldap3.Server(host=self.ldap_server_url,
+								  tls=tls_configuration)
+			bind_type = ldap3.AUTO_BIND_TLS_BEFORE_BIND if self.ssl_tls_mode == "StartTLS" else True
+
+			conn = ldap3.Connection(server=server,
+									user=base_dn,
```

Comment:
fix indentations!

Reply:
Hi saurabah, all of those indentations seem to be a github issue, if you open the file, the tabs are correct, I've tried it in a couple of different editors and they are all consistent...
![image](https://user-images.githubusercontent.com/3469512/59273925-d695e000-8c8b-11e9-892b-0ad2c6b41e40.png)
![image](https://user-images.githubusercontent.com/3469512/59273958-e6152900-8c8b-11e9-9109-54bd9c7da294.png)

Reply:
I've redone all the tabs, and checked in an editor with 2 space tabs vs 4 space tabs and seems to be consistent now...wow what a pain! - at least I know what will need to do in the future to get it to come out ok.